### ISSUE TITLE
Deal dashboard should redirect to the homepage for unauthorized private deal 

### ISSUE THREAD

**Issue by giladt** [PrimeDAO/prime-deals-dapp#387]
If a deal is private and no wallet is connected or if the connected wallet address is not authorized, users should be redirected to the home page.

**Comment by curvelabs-dev** [comment]
Seems like there is a corollary here:  The user should not be able to get to the dashboard to begin with in this case (via the UI).

**Comment by giladt** [comment]
> Seems like there is a corollary here: The user should not be able to get to the dashboard to begin with in this case (via the UI).

True, unless an unauthorized user is trying to access the page directly using a URL.
